src/datachain/lib/arrow.py

```diff
@@ -43,13 +44,17 @@ def __init__(
         self.kwargs = kwargs
 
     def process(self, file: File):
-        path = file.get_path()
-        ds = dataset(
-            path, filesystem=file.get_fs(), schema=self.input_schema, **self.kwargs
-        )
+        if self.nrows:
+            path = _nrows_file(file, self.nrows)
+            ds = dataset(path, schema=self.input_schema, **self.kwargs)
+        else:
+            path = file.get_path()
+            ds = dataset(
+                path, filesystem=file.get_fs(), schema=self.input_schema, **self.kwargs
+            )
         index = 0
         with tqdm(desc="Parsed by pyarrow", unit=" rows") as pbar:
-            for record_batch in ds.to_batches(use_threads=False):
```

Comment:
why removing use_threads?

Reply:
`use_threads=False` was a hack to make nrows not hang forever. Previously nrows broke the batch iteration and hung forever without `use_threads=False`. We no longer are messing with the batch iteration, so this isn't needed.